modules/InboundEmail/InboundEmail.php

```diff
@@ -504,6 +495,27 @@ function($a, $b) use($sortCRM) {  // defaults to DESC order
             "mailbox_info" => json_decode(json_encode($mailboxInfo), true),
         );
     }
+
+    /**
+     * @param $imapStructure
+     * @return bool
+     */
+    private function mesageStructureHasAttachment($imapStructure)
+    {
+        if (isset($imapStructure->parts)) {
+            foreach ($imapStructure->parts as $part) {
+                if ($this->mesageStructureHasAttachment($part)) {
+                    return true;
+                }
+            }
+        } else {
+            if (isset($imapStructure->disposition) && $imapStructure->disposition == 'attachment') {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
You can split the if condition, like:

`        if (!isset($imapStructure->parts)
            && isset($imapStructure->disposition)
            && $imapStructure->disposition == 'attachment') {
            return true;
        }

        if (isset($imapStructure->parts)) {
            foreach ($imapStructure->parts as $part) {
                if ($this->mesageStructureHasAttachment($part)) {
                    return true;
                }
            }
        }

        return false;
`

I believe it's more readable and you return a case immediately

Reply:
Yes I agree, makes it easier to read.